src/Agent.Listener/Configuration/WindowsSecurityManager.cs

```diff
@@ -0,0 +1,477 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Cryptography;
+using System.Security.Principal;
+using System.Text;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator (Default = typeof(WindowsSecurityManager))]
+    public interface IWindowsSecurityManager : IAgentService
+    {
+        string GetUniqueBuildGroupName();
+
+        bool LocalGroupExists(string groupName);
+
+        void CreateLocalGroup(string groupName);
+
+        void AddMemberToLocalGroup(string accountName, string groupName);
+
+        void GrantFullControlToGroup(string path, string groupName);
+
+        bool GrantUserLogonAsServicePrivilage(string accountName);
+
+        bool IsValidCredential(string logonAccount, string logonPassword);
+
+        NTAccount GetDefaultServiceAccount();
+
+        void SetPermissionForAccount(string path, string accountName);
+    }
+
+    public class WindowsSecurityManager : AgentService, IWindowsSecurityManager
+    {
+        private const string AgentServiceLocalGroupPrefix = "VSTS_AgentService_G";
+
+        public string GetUniqueBuildGroupName()
+        {
+            return AgentServiceLocalGroupPrefix + GetHashCodeForAgent().Substring(0, 5);
+        }
+
```

Comment:
The code flow is mostly identical except for install and unstall additional win32 apis are invoked.

In the old agent the username/domainname/logonname were cut/split where ever required. 
Now its all done in one place right after reading the logonusername.

There are some pieces are missing or marked as TODO in the new agent code, 
 - checking for the directory has permission before assigning acl to the work directory, the reason I see some types are missing in coreclr. I had to find another way to achieve this.
 - If its domain joined machine the networkservice is used instead of localservice, again there are some types are missing to do the same in the new agent.
 - If localservice is used the account is added to corresponding pool group, which is yet to be done.
 - Adding Service description/RegKeys, marked as TODO.